src/mmw/js/src/core/modals/views.js

```diff
@@ -55,16 +55,22 @@ var ConfirmView = ModalBaseView.extend({
     template: modalConfirmTmpl,
 
     ui: {
-        confirmation: '.confirm'
+        confirmation: '.confirm',
+        deny: '.btn-default'
     },
 
     events: _.defaults({
-        'click @ui.confirmation': 'primaryAction'
+        'click @ui.confirmation': 'primaryAction',
+        'click @ui.deny': 'cancelAction'
     }, ModalBaseView.prototype.events),
 
     primaryAction: function() {
         this.triggerMethod('confirmation');
         this.hide();
+    },
+
+    cancelAction: function() {
```

Comment:
A similar method exists in `user/views.js` where it is called `dismissAction`.

This function only works when the user clicks cancel. If they click outside the modal, or use <kbd>Escape</kbd>, then the app is in an inconsistent state where neither event was triggered:

![mmw-clear-aoi-issue](https://cloud.githubusercontent.com/assets/1430060/8937476/b2872590-3526-11e5-9c33-8ddf247bf20f.gif)

Consider refactoring so that instead of listening to the `deny` event, the consumer listens for the `hidden.bs.modal` and checks for a flag that indicates if `primaryAction` was executed or not.